src/main/java/io/jenkins/stapler/idea/jelly/IconSrcCompletionContributor.java

```diff
@@ -0,0 +1,56 @@
+package io.jenkins.stapler.idea.jelly;
+
+import com.intellij.codeInsight.completion.CompletionContributor;
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionProvider;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.CompletionType;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.project.Project;
+import com.intellij.patterns.PlatformPatterns;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.impl.source.xml.XmlAttributeImpl;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.util.ProcessingContext;
+import io.jenkins.stapler.idea.jelly.symbols.SymbolFinder;
+import org.jetbrains.annotations.NotNull;
+
+public class IconSrcCompletionContributor extends CompletionContributor {
+
+    public IconSrcCompletionContributor() {
+        extend(
+                CompletionType.BASIC,
+                PlatformPatterns.psiElement().inside(XmlAttributeImpl.class),
+                new CompletionProvider<>() {
+                    @Override
+                    protected void addCompletions(
+                            @NotNull CompletionParameters parameters,
+                            @NotNull ProcessingContext context,
+                            @NotNull CompletionResultSet result) {
+                        PsiElement position = parameters.getPosition();
+                        PsiElement parent = position.getParent().getParent();
+
+                        if (isInsideLIconSrcAttribute(parent)) {
+                            Project project = position.getProject();
+                            SymbolFinder.getInstance(project)
+                                    .getAvailableSymbols()
+                                    .forEach(file -> result.addElement(LookupElementBuilder.create(file.name())
+                                            .withPresentableText(file.displayText())
+                                            .withTypeText(file.group())));
+                        }
```

Comment:
One annoying bug in this, if you have an existing symbol, e.g.

```xml
<l:icon src="symbol-app-bar plugin-design-library" />
```

Then you change (using auto-completion) to `symbol-buttons`
You get the plugin doubled up:

```xml
<l:icon src="symbol-buttons plugin-design-library plugin-design-library" />
```

You get a similar problem if you are now getting it from a different plugin:

```xml
<l:icon src="symbol-apps-outline plugin-ionicons-api plugin-design-library" />
```

Would be good to fix this

Reply:
Good idea, updated it with an `.insertHandler(...)`